cmd/storer2elastic/dataprocessor/dataReplayer_test.go

```diff
@@ -0,0 +1,307 @@
+
+	handlerFunc := func(persistedData data.RoundPersistedData) bool {
+		return true
+	}
+	err := dr.Range(handlerFunc)
+	require.Equal(t, dataprocessor.ErrNoMetachainDatabase, err)
+}
+
+func TestDataReplayer_Range_EpochStartMetaBlockFoundForEpochStartKeyShouldWork(t *testing.T) {
+	t.Parallel()
+
+	marshalizer := &mock.MarshalizerMock{}
+
+	shard0HdrHash := []byte("sh0Hdr")
+	txMbHash := []byte("txMbHash")
+	rwdTxMbHash := []byte("txMbHash")
+	uTxMbHash := []byte("uTxMbHash")
+	txHash := []byte("txHash")
+	rwdTxHash := []byte("rwdTx")
+	uTxHash := []byte("uTxHash")
+
+	epStMb := &block.MetaBlock{Nonce: 0, Epoch: 0, ShardInfo: []block.ShardData{{HeaderHash: shard0HdrHash}}}
+	epStMbBytes, _ := marshalizer.Marshal(epStMb)
+	metaBlocksPersister := mock.NewPersisterMock()
+	_ = metaBlocksPersister.Put([]byte("epochStartBlock_0"), epStMbBytes)
+
+	sh0Hdr := &block.Header{
+		Round: 0,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{Type: block.TxBlock, SenderShardID: core.MetachainShardId, ReceiverShardID: 0, Hash: txMbHash},
+			{Type: block.RewardsBlock, SenderShardID: core.MetachainShardId, ReceiverShardID: 0, Hash: rwdTxMbHash},
+			{Type: block.SmartContractResultBlock, SenderShardID: core.MetachainShardId, ReceiverShardID: 0, Hash: uTxMbHash},
+		},
+	}
+	sh0HdrBytes, _ := marshalizer.Marshal(sh0Hdr)
+	shardHeadersPersister := mock.NewPersisterMock()
+	_ = shardHeadersPersister.Put(shard0HdrHash, sh0HdrBytes)
+
+	miniBlocksPersister := mock.NewPersisterMock()
+	mbHdr := &block.MiniBlock{Type: block.TxBlock, TxHashes: [][]byte{txHash}}
+	mbHdrBytes, _ := marshalizer.Marshal(mbHdr)
+	_ = miniBlocksPersister.Put(txMbHash, mbHdrBytes)
+
+	rwdMbHdr := &block.MiniBlock{Type: block.RewardsBlock, TxHashes: [][]byte{rwdTxHash}}
+	rwdMbHdrBytes, _ := marshalizer.Marshal(rwdMbHdr)
+	_ = miniBlocksPersister.Put(rwdTxMbHash, rwdMbHdrBytes)
+
+	uMbHdr := &block.MiniBlock{Type: block.SmartContractResultBlock, TxHashes: [][]byte{uTxHash}}
+	uMbHdrBytes, _ := marshalizer.Marshal(uMbHdr)
+	_ = miniBlocksPersister.Put(uTxMbHash, uMbHdrBytes)
+
+	txPersister := mock.NewPersisterMock()
+	tx := &transaction.Transaction{Nonce: 37}
+	txBytes, _ := marshalizer.Marshal(tx)
+
+	rwdTx := rewardTx.RewardTx{Round: 37}
+	rwdTxBytes, _ := marshalizer.Marshal(rwdTx)
+
+	uTx := smartContractResult.SmartContractResult{Nonce: 37}
+	uTxBytes, _ := marshalizer.Marshal(uTx)
+
+	_ = txPersister.Put(txHash, txBytes)
+	_ = txPersister.Put(rwdTxHash, rwdTxBytes)
+	_ = txPersister.Put(uTxHash, uTxBytes)
+
+	args := getDataReplayArgs()
+	args.DatabaseReader = &mock.DatabaseReaderStub{
+		GetDatabaseInfoCalled: func() ([]*databasereader.DatabaseInfo, error) {
+			return []*databasereader.DatabaseInfo{
+				{
+					Epoch: 0,
+					Shard: core.MetachainShardId,
+				},
+				{
+					Epoch: 0,
+					Shard: 0,
+				},
+			}, nil
+		},
+		GetStaticDatabaseInfoCalled: func() ([]*databasereader.DatabaseInfo, error) {
+			return []*databasereader.DatabaseInfo{
+				{
```

Comment:
maybe this could be replaced with :
`return getDbInfoMetaAndOneShard(), nil`

Reply:
done